### ISSUE TITLE
LayerNorm, what is going on?

### ISSUE THREAD

**Issue by ju-w** [facebookresearch/ConvNeXt#136]
Hello!

I am unsure of what the LayerNorm on images is actually supposed to do. **LayerNorm channels first** works kinda like **BatchNorm2d**, however with quite suspicious vertical lines. **LayerNorm channels last** however completely breaks the image, as each _pixel_ is normalized individually. Is this intended behaviour?

Please have a look at the experiment below, to understand what I mean.

Related  #112 possibly #115

``` python
import torch
from torch import nn
import torchvision
from models.convnext import LayerNorm


img = torchvision.io.read_image("cat.jpg")
img = img*0.005
torchvision.utils.save_image(img, "cat0.jpg")
```

![cat](https://user-images.githubusercontent.com/22564375/209563993-c81301a0-e420-4d47-a7ea-fe037483be1d.jpg)

# Batch Norm 2d
```python
bn = nn.BatchNorm2d(3)
img_bn = bn(img.unsqueeze(0))[0]
torchvision.utils.save_image(img_bn*.28+.7, "cat_bn.jpg")
```
![cat_bn](https://user-images.githubusercontent.com/22564375/209564002-36a0393d-72f6-491e-a34b-8a86dbe7463f.jpg)

# Layer Norm
## channels first
```python
ln_cf = LayerNorm(3, data_format="channels_first")
img_cf = ln_cf(img)
torchvision.utils.save_image(img_cf*.28+.7, "cat_cf.jpg")
```
![cat_cf](https://user-images.githubusercontent.com/22564375/209564005-589c88e3-4ef7-4dca-a0db-7850d677dcd5.jpg)

## channels last
```python
ln_cl = LayerNorm(3, data_format="channels_last")
img_cl = ln_cl(img.permute(1,2,0)).permute(2,0,1)
torchvision.utils.save_image(img_cl*.28+.7, "cat_cl.jpg")
```
![cat_cl](https://user-images.githubusercontent.com/22564375/209564011-103b356d-58f5-4ecc-9ed7-9b03f975c713.jpg)

**Comment by ju-w** [comment]
See also: pytorch/pytorch#51455
